### PR TITLE
Feature/error checking apis

### PR DESCRIPTION
**Pull request by BaldwinJordan**

### CODE REVIEW

**Comment by BaldwinJordan** [comment]
Requires fsbid changes before review

File: talentmap_api/fsbid/services/common.py [hunk 1]
```diff
@@ -129,13 +132,19 @@ def get_results(uri, query, query_mapping_function, jwt_token, mapping_function)
     url = f"{API_ROOT}/{uri}?{query_mapping_function(query)}"
     response = requests.get(url, headers={'JWTAuthorization': jwt_token, 'Content-Type': 'application/json'}, verify=False).json()  # nosec
 
-    return list(map(mapping_function, response["Data"]))
+    if response.get('return_code', -1) == -1:
```
Comment (rtirserio) [inline talentmap_api/fsbid/services/common.py]: Check for `response.get("Data") is None` here in addition to the return_code value

File: talentmap_api/fsbid/services/common.py [hunk 2]
```diff
@@ -129,13 +132,19 @@ def get_results(uri, query, query_mapping_function, jwt_token, mapping_function)
...
+        return None
+
+    return list(map(mapping_function, response.get("Data", "")))
```
Comment (rtirserio) [inline talentmap_api/fsbid/services/common.py]: The fallback to `response.get("Data")` should be `{}`

File: talentmap_api/fsbid/services/common.py [hunk 3]
```diff
@@ -129,13 +132,19 @@ def get_results(uri, query, query_mapping_function, jwt_token, mapping_function)
...
 
 def get_fsbid_results(uri, jwt_token, mapping_function):
     url = f"{API_ROOT}/{uri}"
     response = requests.get(url, headers={'JWTAuthorization': jwt_token, 'Content-Type': 'application/json'}, verify=False).json() # nosec
     
-    return map(mapping_function, response["Data"])
+    if response.get('return_code', -1) == -1:
```
Comment (rtirserio) [inline talentmap_api/fsbid/services/common.py]: See above

**Comment by rtirserio** [comment]
@BaldwinJordan the available_positions.py file still has the default set to ""

**Pull request merged**